cmd/ttn-lw-cli/commands/applications_access.go

```diff
@@ -263,11 +251,32 @@ var (
 				return errNoAPIKeyRights
 			}
 
-			res, err := createApplicationAPIKey(ctx, *appID, name, rights...)
+			expiry, _ := cmd.Flags().GetString("api-key-expiry")
+			var expiryDate time.Time
+
+			if expiry != "" {
+				expiryDate, err := time.Parse("2006-01-02", expiry)
+				if err != nil {
+					return errInvalidDateFormat
+				}
+				if expiryDate.Before(time.Now()) {
+					return errExpiryDateInPast
+				}
+			}
```

Comment:
- You do this quite often, maybe extract it into a func?
- Should we take midnight (the beginning of the day) or a minute to midnight (the end of the day)?
- What timezone is used?

Reply:
>What timezone is used?

the timezone is UTC, since it's the server time. should we make it timezone aware? I can think of some edge cases in the current implementation where the api key could expire before the user expects it to

>Should we take midnight (the beginning of the day) or a minute to midnight (the end of the day)?

We could also take the minute exactly from the api key creation?

Reply:
The server indeed uses UTC, but clients (like also the Console) may use local time.

After giving this some more thought I think we should just let users pass the entire date+time instead of only the date; that way they can decide.